src/playlist-maker.js

```diff
@@ -78,6 +78,46 @@ var playlistMaker = function(player, plist) {
     return currentIndex;
   };
 
+  // item can be either
+  //  * a string
+  //  * an array of sources, which are either strings or {src, type} objects
+  //  * a playlist item
+  playlist.contains = function contains(item) {
+    return player.playlist.indexOf(item) !== -1;
+  };
+
+  playlist.indexOf = function indexOf(item) {
+    var ret = -1;
+    var sources;
+    var source;
+    var i;
+
+    if (typeof item === 'string') {
+      ret = indexInSources(list, item);
+    } else {
```

Comment:
My personal preference is to return early if the other half of the conditional is more than a line or two so the indentation is easier to parse. Just a thought, no action necessary.